microsoft/botframework-solutions
# [Botskills] Implement utils tests

### Purpose
*What is the context of this pull request? Why is it being done?*
There were no tests for `utils` implementation in `Botskills` cli tool

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

- Implement new tests which validate the `utils` implementation
- Update the error message of authentication process making it as informative one
- Remove unused (`spawn`) method
- Undo changes on temporal test files during the execution of the tests
- Fix typos

![image](https://user-images.githubusercontent.com/11904023/61547999-fe166000-aa22-11e9-9a25-b6f58118df90.png)


### Tests
*Is this covered by existing tests or new ones? If no, why not?*
We added the following tests:
- `authentication.test.js`
- `childProcess.test.js`
- `sanitization.test.js`
- `validation.test.js`

### Testing Steps
1. Go to `.\botframework-solutions\tools\botskills`
1. Execute `npm install`
1. Execute `npm run build`
1. Execute `npm run test`
1. Check that all the tests are passing successfully

![image](https://user-images.githubusercontent.com/11904023/61547933-cb6c6780-aa22-11e9-90d5-eb9eb7be5e75.png)

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

### Checklist

#### General

#### Bots

#### Deployment Scripts
